.github/actions/generate-coverage/scripts/run_rust.py

```
@@ -341,48 +369,137 @@ def _pump_cargo_output(proc: subprocess.Popen[str]) -> list[str]:
+    message = f"cargo output streams not captured: missing {missing}"
+    with contextlib.suppress(Exception):
+        proc.kill()
+    with contextlib.suppress(Exception):
+        proc.wait(timeout=5)
+    _safe_close_text_stream(typ.cast("typ.TextIO | None", proc.stdout))
+    _safe_close_text_stream(typ.cast("typ.TextIO | None", proc.stderr))
+    typer.echo(f"::error::{message}", err=True)
+    raise typer.Exit(1) from None
+
+
+def _wait_for_cargo(proc: subprocess.Popen[str], args: list[str]) -> int:
```

Ruff (0.15.10)

[warning] Unused function argument: `args`

(ARG001)